### PR TITLE
A lot of changes make it into NATIVESVTTK

### PR DESCRIPTION
**Pull request by jhi2**
**IMPORTANT: TITLEBAR SUPPORT USES WINDOWS RUNTIMES AND FEATURES THEREFORE MAKING IT WINDOWS-ONLY!**
Requires darkdetect for native theme support

### CODE REVIEW

File: sv_ttk/__init__.py [hunk 1]
```diff
@@ -42,7 +44,44 @@ def toggle_theme(root: tkinter.Tk | None = None) -> None:
     _load_theme(style)
 
     set_theme("light" if style.theme_use() == "sun-valley-dark" else "dark")
-
+def titlebar(root: tkinter.tk,theme: "light"):
```
Comment (jhi2) [inline sv_ttk/__init__.py:47]: **IMPORTANT: TITLEBAR SUPPORT ONLY WORKS ON WINDOWS!!!!!!!!!!!!!!!!!!!!!!!!!!!**

File: sv_ttk/__init__.py [hunk 2]
```diff
@@ -42,7 +44,44 @@ def toggle_theme(root: tkinter.Tk | None = None) -> None:
...
+    value = ct.c_int(value)
+    set_window_attribute(hwnd, rendering_policy, ct.byref(value), ct.sizeof(value))
+if theme not in {"dark", "light"}:
+        raise RuntimeError(f"not a valid sv_ttk theme: {theme}")
+else:
+    if theme == "light":
+        light_title_bar(root)
+    elif theme == "dark":
+        dark_title_bar(root)
+def native_theme(root:tkinter.tk):
```
Comment (jhi2) [inline sv_ttk/__init__.py:75]: **NATIVE THEME SUPPORT INCLUDES TITLEBAR SUPPORT THUS MAKING IT WINDOWS-ONLY!!!!!!**

File: sv_ttk/__init__.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
 from functools import partial
 from pathlib import Path
 from tkinter import ttk
+import ctypes as ct
+import darkdetect
```
Comment (jhi2) [inline sv_ttk/__init__.py:8]: **PACKAGE INSTALL REQUIRED**

File: sv_ttk/__init__.py [hunk 4]
```diff
@@ -42,7 +46,44 @@ def toggle_theme(root: tkinter.Tk | None = None) -> None:
     _load_theme(style)
 
     set_theme("light" if style.theme_use() == "sun-valley-dark" else "dark")
-
+def titlebar(root: tkinter.tk,theme):
```
Comment (jhi2) [inline sv_ttk/__init__.py:49]: **TITLEBAR SUPPORT USES WINDOWS FEATURES, THUS MAKING IT WINDOWS-ONLY **

**Comment by jhi2** [comment]
REALLY COOL! run uploadtopypi/upload.py to upload it to pypi

**Comment by jhi2** [comment]
FYI: I based this off my fork on pypi

**Review by rdbende — CHANGES_REQUESTED** [review]
Hi!
I don't understand what's your goal here. Is this a personal fork of the project and which you accidentally made a PR for, or do you want me to merge this?
If the latter is the case, then I have a few comments, because I'm not going to merge this as is.

1. I don't want do depend on neither `darkdetect` nor `pillow`. There's a reason why there's no function for setting the system theme, if people need it, they can always pass in the result provided by `darkdetect.theme()`. Pillow isn't needed either, because Tkinter is able to load PNG images by default.
2. Don't use `os.system`. If you really want to call shell commands, then use `subprocess.call` or `subprocess.Popen` depending on your needs, but in this case it's really isn't needed. If you need to install a dependency, put it in `requirements.txt` or `pyproject.toml`, but never use a `pip install` call, because people might not have pip installed at all, and that will make the whole package unable to load.
3. Speaking of pip, I already have (or just **had**, because you deleted it) an automated script, that uploads the package to Pypi, so there's no need to introduce a Python script for that. And btw, you exposed your Pypi token. Generate a new one ASAP, because now anyone is able to upload packages to your account using it. **NEVER EVER COMMIT SUCH THINGS TO VERSION CONTROL!!!**
4. Also, please don't restructure the whole project. I'm happy with the current structure, and I don't need new directories, unnecessary files, or a rewritten README. Adding _"This is a pull request based off of sv-ttk theme by rdbende"_ is pointless, because when it is merged into main, it will be completely irrelevant whose project this particular PR was based upon, and it will be clear that I am the project owner.
5. Those cool messageboxes are not littlewhitecould's, but they were written by me. I want to include that code in this package eventually, but it's incomplete, that's why it lives in [this separate repo](https://github.com/rdbende/Sun-Valley-messageboxes)
6. Don't change the licence file. It doesn't need to include all contributors, and especially don't overwrite my email address, unless you want to get that many support emails that I get :D
7. If you revert this PR to just adding the dark mode titlebars feature, I think it's fine, but please don't comment about how Windows-only it is, but simply add am `if sys.platform == "win32"`, to make it sure, that it will run only on Window. Additionally, you will need to add an attribution to the [original code source and author](https://gist.github.com/Olikonsti/879edbf69b801d8519bf25e804cec0aa) on the top of the file.

**Comment by jhi2** [comment]
Try this

**Pull request closed without merging**